Engine/BaseAssets/Shaders/Bloom/bloom_downsample_box.fsh

```diff
@@ -0,0 +1,36 @@
+struct vert_in
+{
+    float4 _ : SV_POSITION;
+    float2 t : TEX;
+};
+
+cbuffer textureBuf
+{
+    Texture2D tex;
+};
+
+cbuffer textureParams
+{
+    float2 texSize;
+};
+
+SamplerState texSampler;
+
+float3 box(float2 uv, float2 texelSize, float delta)
+{
+    float3 color;
+    
+    color = tex.Sample(texSampler, uv + float2(texelSize.x, texelSize.y)).rgb;
+    color += tex.Sample(texSampler, uv + float2(-texelSize.x, texelSize.y)).rgb;
+    color += tex.Sample(texSampler, uv + float2(texelSize.x, -texelSize.y)).rgb;
+    color += tex.Sample(texSampler, uv + float2(-texelSize.x, -texelSize.y)).rgb;
+    
+    return color * 0.25f;
+}
+
+float4 main(vert_in v) : SV_Target
+{
+    float2 texelSize = float2(1, 1) / texSize;
```

Comment:
precalculate this in c#, so shader has 1 less division to do for every pixel

Reply:
Done